app/jobs/maestrano/connector/rails/push_to_connec_job.rb

```diff
@@ -11,35 +11,35 @@ def perform(organization, entities_hash, opts={})
       last_synchronization = organization.last_successful_synchronization
 
       entities_hash.each do |external_entity_name, entities|
-        if entity_instance_hash = find_entity_instance(external_entity_name)
+        if entity_instance_hash = find_entity_instance(external_entity_name, organization, connec_client, external_client)
           next unless organization.synchronized_entities[entity_instance_hash[:name].to_sym]
 
           entity_instance = entity_instance_hash[:instance]
 
-          entity_instance.before_sync(connec_client, external_client, last_synchronization, organization, opts)
+          entity_instance.before_sync(last_synchronization)
           # Build expected input for consolidate_and_map_data
           if entity_instance_hash[:is_complex]
-            mapped_entities = entity_instance.consolidate_and_map_data(Hash[ *entity_instance.class.connec_entities_names.collect{|name| [ name, []]}.flatten(1) ], Hash[ *entity_instance.class.external_entities_names.collect{|name| name == external_entity_name ? [name, entities] : [ name, []]}.flatten(1) ], organization, {})
+            mapped_entities = entity_instance.consolidate_and_map_data(Hash[ *entity_instance.class.connec_entities_names.collect{|name| [ name, []]}.flatten(1) ], Hash[ *entity_instance.class.external_entities_names.collect{|name| name == external_entity_name ? [name, entities] : [ name, []]}.flatten(1) ])
```

Comment:
please add some comment on what this lovely line is doing, or extract it to a private static method that describe the operation.